packages/user/src/supertokens/recipes/config/third-party-email-password/emailPasswordSignIn.ts

```diff
@@ -27,33 +25,37 @@ const emailPasswordSignIn = (
       return originalResponse;
     }
 
-    const service: UserProfileService<
-      UserProfile & QueryResultRow,
-      UserProfileCreateInput,
-      UserProfileUpdateInput
-    > = new UserProfileService(config, slonik);
+    const userService: UserService<
+      User & QueryResultRow,
+      UserCreateInput,
+      UserUpdateInput
+    > = new UserService(config, slonik);
 
-    /* eslint-disable-next-line unicorn/no-null */
-    let profile: UserProfile | null = null;
+    let user: User | null | undefined;
 
     try {
-      profile = await service.findById(originalResponse.user.id);
+      user = await userService.update(originalResponse.user.id, {
+        lastLoginAt: new Date()
```

Comment:
can you extract the logic of formatting date into a utility function so that it can be reuse in other places as well?